app/src/main/java/org/mozilla/fenix/settings/address/ext/Address.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address.ext
+
+import mozilla.components.concept.storage.Address
+
+/**
+ * Generate a label from a given [Address]. A well-formed [Address] will have
+ * a full name in the label, otherwise it will find the highest priority [Address] property
+ * to use, as based on the desktop code found in: FormAutofillUtils::getAddressLabel
+ */
+fun Address.toLabel(): String = getName().ifEmpty { iterateFields("") }
+
+/**
+ * Generate a description from a given [Address]. A well-formed address will have a street address
+ * as a description. An address with missing data will use the next available property after [label]
+ * based on the priorities found in the desktop code in: FormAutofillUtils::getAddressLabel
+ */
+fun Address.toDescription(label: String): String = iterateFields(label)
+
+private fun Address.getName(): String {
+    val middleName = if (additionalName.isEmpty()) "" else "$additionalName "
+    return if (givenName.isNotEmpty()) "$givenName $middleName$familyName" else familyName
+}
+
+private fun Address.iterateFields(previous: String): String =
+    streetAddress.joinStreetAddress()
+        .ifEmptyOrPrevious(previous) { addressLevel3 }
+        .ifEmptyOrPrevious(previous) { addressLevel2 }
+        .ifEmptyOrPrevious(previous) { organization }
+        .ifEmptyOrPrevious(previous) { addressLevel1 }
+        .ifEmptyOrPrevious(previous) { country }
+        .ifEmptyOrPrevious(previous) { postalCode }
+        .ifEmptyOrPrevious(previous) { tel }
+        .ifEmptyOrPrevious(previous) { email }
+        .ifEmptyOrPrevious(previous) { "" }
+
+private fun String.joinStreetAddress(): String =
+    this.split("\n").joinToString(separator = " ") { it.trim() }
```

Comment:
This will trim the split parts. `joinToString` will join the result of the block defined here as `it.trim` with the `separator` parameter.

Reply:
I know we have a test case for something that is like this already, but I think it also makes sense to just call this function individually since I found a whole suite of test cases for this function alone in https://searchfox.org/mozilla-central/source/browser/extensions/formautofill/test/unit/test_toOneLineAddress.js.

It might make sense to also just use the same naming as desktop.

Reply:
Yeah, I can do that. I only structured it that way because I generally try to avoid `@VisibleForTesting`, especially in single module projects.